src/pyhf/infer/calculators.py

```diff
@@ -574,6 +675,110 @@ def distributions(self, poi_test, track_progress=None):
+            b_only_distribution (~pyhf.infer.calculators.EmpiricalDistribution):
+              The distribution for the background-only hypothesis.
+
+        Returns:
+            Tuple (:obj:`float`): The :math:`p`-values for the test statistic
+            corresponding to the :math:`\mathrm{CL}_{s+b}`,
+            :math:`\mathrm{CL}_{b}`, and :math:`\mathrm{CL}_{s}`.
+        """
+        CLsb = sig_plus_bkg_distribution.pvalue(teststat)
+        CLb = b_only_distribution.pvalue(teststat)
+        CLs = CLsb / CLb
+        return CLsb, CLb, CLs
+
+    def expected_pvalues(self, sig_plus_bkg_distribution, b_only_distribution):
+        r"""
+        Calculate the :math:`\mathrm{CL}_{s}` values corresponding to the
+        median significance of variations of the signal strength from the
+        background only hypothesis :math:`\left(\mu=0\right)` at
+        :math:`(-2,-1,0,1,2)\sigma`.
+
+        Example:
+
+            >>> import pyhf
+            >>> import numpy.random as random
+            >>> random.seed(0)
+            >>> pyhf.set_backend("numpy")
+            >>> model = pyhf.simplemodels.hepdata_like(
+            ...     signal_data=[12.0, 11.0], bkg_data=[50.0, 52.0], bkg_uncerts=[3.0, 7.0]
+            ... )
+            >>> observations = [51, 48]
+            >>> data = observations + model.config.auxdata
+            >>> mu_test = 1.0
+            >>> toy_calculator = pyhf.infer.calculators.ToyCalculator(
+            ...     data, model, ntoys=100, track_progress=False
+            ... )
+            >>> sig_plus_bkg_dist, bkg_dist = toy_calculator.distributions(mu_test)
+            >>> CLsb_exp_band, CLb_exp_band, CLs_exp_band = toy_calculator.expected_pvalues(sig_plus_bkg_dist, bkg_dist)
+            >>> CLs_exp_band
+            [0.0, 0.0, 0.06186224489795918, 0.2845003327965815, 1.0]
+
+        Args:
+            sig_plus_bkg_distribution (~pyhf.infer.calculators.EmpiricalDistribution):
+              The distribution for the signal + background hypothesis.
+            b_only_distribution (~pyhf.infer.calculators.EmpiricalDistribution):
+              The distribution for the background-only hypothesis.
+
+        Returns:
+            Tuple (:obj:`float`): The :math:`p`-values for the test statistic
+            corresponding to the :math:`\mathrm{CL}_{s+b}`,
+            :math:`\mathrm{CL}_{b}`, and :math:`\mathrm{CL}_{s}`.
+        """
+        tb, _ = get_backend()
+        pvalues = tb.astensor(
+            [
+                self.pvalues(
+                    tb.astensor(test_stat),
+                    sig_plus_bkg_distribution,
+                    b_only_distribution,
+                )
+                for test_stat in b_only_distribution.samples
+            ]
+        )
+        # TODO: Add percentile to tensorlib
+        # c.f. Issue #815, PR #817
+        import numpy as np
+
+        _percentiles = [2.27501319, 15.86552539, 50.0, 84.13447461, 97.72498681]
```

Comment:
I think we're going to need a comment here or some better description in the docstring as the nature of these percentiles as it is late and I don't remember what exactly these are &mdash; which mens there will be a time in the future when it will also be late and I will want to know what these are.

Reply:
Oh right, as this is for the background only hypothesis then these are Normal percentiles.